tests/roots.rs

```diff
@@ -0,0 +1,84 @@
+extern crate num_bigint;
+extern crate num_integer;
+extern crate num_traits;
+
+mod biguint {
+    use num_bigint::BigUint;
+    use num_traits::FromPrimitive;
+    use std::str::FromStr;
+
+    fn check(x: i32, n: u32, expected: i32) {
+        let big_x: BigUint = FromPrimitive::from_i32(x).unwrap();
```

Comment:
If you use an unsigned type instead of `i32`, then `From` makes this simpler -- `BigUint::from(x)`.